packages/backend/src/database/migrations/20241108145015_group-names-unique.ts

```diff
@@ -0,0 +1,44 @@
+import { Knex } from 'knex';
+
+const GroupsTableName = 'groups';
+
+export async function up(knex: Knex): Promise<void> {
+    // Ensure no duplicate names for organization_id by appending suffixes
+    const duplicates = await knex(GroupsTableName)
+        .select('name', 'organization_id')
+        .count('* as duplicate_count')
+        .groupBy('name', 'organization_id')
+        .having(knex.raw('count(*) > ?', [1]));
+
+    for (const duplicate of duplicates) {
+        // eslint-disable-next-line @typescript-eslint/naming-convention
+        const { name, organization_id } = duplicate;
+
+        // Get all rows with this name and organization_id to rename them
+        // eslint-disable-next-line no-await-in-loop
+        const rows = await knex(GroupsTableName)
+            .select('group_uuid')
+            .where({ name, organization_id })
+            .orderBy('created_at', 'asc');
+
+        // Rename duplicates with suffixes
+        // eslint-disable-next-line no-plusplus
+        for (let i = 1; i < rows.length; i++) {
+            // eslint-disable-next-line no-await-in-loop
+            await knex(GroupsTableName)
+                .where({ group_uuid: rows[i].group_uuid })
+                .update({ name: `${name} ${i}` });
```

Comment:
If an org has 2 groups named the same "Testing" it will migrate them "Testing 1" and "Testing 2"

Reply:
Correction. In the case of two groups with the same name, the first one keeps the original while the rest gets a numeric suffix.
before migration
<img width="1035" alt="Screenshot 2024-11-11 at 10 07 44" src="https://github.com/user-attachments/assets/8f117aae-941b-4c05-97c3-f64730ed57c5">

after migration
<img width="1035" alt="Screenshot 2024-11-11 at 10 08 25" src="https://github.com/user-attachments/assets/0b6342d9-a78f-417d-93d9-8c0cf5a5cea2">

Reply:
You are correct. I've modified it so that the original stays the same and the next one gets "Name 2" and so on